### PR TITLE
docs: add documentation for starting with bzlmod

### PR DESCRIPTION
**Pull request by alexeagle**
This would be useful for a training course I'm teaching, as well as casual users who are giving this a try.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,10 @@ Support for Bazel versions before `4.0.0` is only available on rules_jvm_externa
 
 ## Usage
 
+> New: on Bazel 6, you can use bzlmod instead of following directions below.
+> See [bzlmod](./docs/bzlmod.md).
+> Expect rough edges and incomplete support as bzlmod is still new as of early 2023.
```
Comment (jin) [inline README.md:73]: > Expect rough edges and incomplete support as bzlmod is still a new feature.

File: docs/bzlmod.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Note: this support is new as of early 2023, so expect some brokenness and missing features.
+Please do file issues for missing bzlmod support.
+
+See the `/examples/bzlmod` folder in this repository for a complete, tested example.
+
+## Installation
+
+First, you must enable bzlmod.
+Note, the Bazel team plans to enable it by default starting in version 7.0.
```
Comment (jin) [inline docs/bzlmod.md:14]: Move this line to the first section with the support information.

File: docs/bzlmod.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+The simplest way is by adding a line to your `.bazelrc` like so:
+```sh
+$ echo common --enable_bzlmod >> .bazelrc
```
Comment (jin) [inline docs/bzlmod.md:17]: Don't use echo (which isn't cross platform), just show a sample .bazelrc file.

File: docs/bzlmod.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+```starlark
+use_repo(maven, "maven", "unpinned_maven")
+```
+
+Now you'll be able to use the same `@unpinned_maven//:pin` operation described in the
+[workspace instructions](/README.md#updating-maven_installjson).
+
+## Known issues
+
+- https://github.com/bazelbuild/rules_jvm_external/issues/827
```
Comment (jin) [inline docs/bzlmod.md:73]: Mind elaborating, instead of links?

**Review by jin — APPROVED** [review]
Please also get @shs96c's review

**Review by shs96c — APPROVED** [review]

**Pull request merged**